### PR TITLE
AEA-1432 Remove 'no-member' and 'unidiomatic-typecheck' pylint errors for generated protocols

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Remove 'no-member' and 'unidiomatic-typecheck' pylint errors for generated protocols

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/protocols/generator/base.py [hunk 1]
```diff
@@ -1521,6 +1527,8 @@ def _serialization_class_str(self) -> str:
             )
         )
 
+        cls_str += "# pylint: disable=no-member\n"
```
Comment (marcofavorito) [inline aea/protocols/generator/base.py:1530]: if added to each line, it does not work well with black

Comment (marcofavorito) [reply aea/protocols/generator/base.py:1530]: image

File: aea/protocols/generator/base.py [hunk 2]
```diff
@@ -64,6 +64,23 @@
...
+    "redefined-builtin",
+]
+PYLINT_DISABLE_MESSAGE_PY = [
+    "too-many-statements",
+    "too-many-locals",
+    "no-member",
+    "too-few-public-methods",
+    "too-many-branches",
+    "not-an-iterable",
+]
```
Comment (marcofavorito) [inline aea/protocols/generator/base.py:81]: https://github.com/fetchai/agents-aea/pull/2419/checks?check_run_id=2208904935#step:5:84

File: Makefile [hunk 3]
```diff
@@ -47,7 +47,7 @@ lint:
 
 .PHONY: pylint
 pylint:
-	pylint -j4 aea benchmark packages scripts plugins/aea-ledger-fetchai/aea_ledger_fetchai plugins/aea-ledger-ethereum/aea_ledger_ethereum plugins/aea-ledger-cosmos/aea_ledger_cosmos plugins/aea-cli-ipfs/aea_cli_ipfs examples/*
+	pylint -j4 aea benchmark packages scripts plugins/aea-ledger-fetchai/aea_ledger_fetchai plugins/aea-ledger-ethereum/aea_ledger_ethereum plugins/aea-ledger-cosmos/aea_ledger_cosmos plugins/aea-cli-ipfs/aea_cli_ipfs examples/* --disable=E1136
```
[the remark this answers is not in the document]
Comment (marcofavorito) [reply Makefile:50]: the same in `tox.ini`

[the remark this answers is not in the document]
Comment (marcofavorito) [reply Makefile:50]: http://pylint-messages.wikidot.com/messages:e1136

File: aea/protocols/generator/base.py [hunk 4]
```diff
@@ -64,6 +64,45 @@
...
+        isinstance(variable_name, variable_type)
+
+    :param variable_name: the variable name.
+    :param variable_type: the variable type.
+    :return: the Python instruction to check the type, in string form.
+    """
+    if variable_type != "bool":
+        return f"isinstance({variable_name}, {variable_type})"
+    else:
+        return f"type({variable_name}) == {variable_type}"
```
Comment (DavidMinarsch) [inline aea/protocols/generator/base.py:103]: why not `f"type({variable_name}) is {variable_type}"?

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**